contexts/AutocratContext.tsx

```diff
@@ -93,122 +85,9 @@ export function AutocratProvider({ children }: { children: ReactNode }) {
-        proposal.account.openbookTwapPassMarket,
-        proposal.account.openbookTwapFailMarket,
-        proposal.account.baseVault,
-        proposal.account.quoteVault,
-      ]);
-      if (!accountInfos || accountInfos.indexOf(null) >= 0) return;
-
-      const pass = await openbook.coder.accounts.decode('market', accountInfos[0]!.data);
-      const fail = await openbook.coder.accounts.decode('market', accountInfos[1]!.data);
-      const passTwap = await openbookTwap.coder.accounts.decodeUnchecked(
-        'TWAPMarket',
-        accountInfos[2]!.data,
-      );
-      const failTwap = await openbookTwap.coder.accounts.decodeUnchecked(
-        'TWAPMarket',
-        accountInfos[3]!.data,
-      );
-      const baseVault = await vaultProgram.coder.accounts.decode(
-        'conditionalVault',
-        accountInfos[4]!.data,
-      );
-      const quoteVault = await vaultProgram.coder.accounts.decode(
-        'conditionalVault',
-        accountInfos[5]!.data,
-      );
-
-      const bookAccountInfos = await connection.getMultipleAccountsInfo([
-        pass.asks,
-        pass.bids,
-        fail.asks,
-        fail.bids,
-      ]);
-      const passAsks = getLeafNodes(
-        await openbook.coder.accounts.decode('bookSide', bookAccountInfos[0]!.data),
-        openbook,
-      );
-      const passBids = getLeafNodes(
-        await openbook.coder.accounts.decode('bookSide', bookAccountInfos[1]!.data),
-        openbook,
-      );
-      const failAsks = getLeafNodes(
-        await openbook.coder.accounts.decode('bookSide', bookAccountInfos[2]!.data),
-        openbook,
-      );
-      const failBids = getLeafNodes(
-        await openbook.coder.accounts.decode('bookSide', bookAccountInfos[3]!.data),
-        openbook,
-      );
-
-      setAllMarketsInfo({
-        ...allMarketsInfo,
-        [proposal.publicKey.toString()]: {
-          pass,
-          passAsks,
-          passBids,
-          fail,
-          failAsks,
-          failBids,
-          passTwap,
-          failTwap,
-          baseVault,
-          quoteVault,
-        },
-      });
-    }, 1000),
-    [allMarketsInfo, vaultProgram, openbook, openbookTwap],
-  );
-  const fetchOpenOrders = useCallback(
-    debounce<[ProposalAccountWithKey, PublicKey]>(
-      async (proposal: ProposalAccountWithKey, owner: PublicKey) => {
-        if (!openbook) {
-          return;
-        }
-        const passOrders = await openbook.account.openOrdersAccount.all([
-          { memcmp: { offset: 8, bytes: owner.toBase58() } },
-          { memcmp: { offset: 40, bytes: proposal.account.openbookPassMarket.toBase58() } },
-        ]);
-        const failOrders = await openbook.account.openOrdersAccount.all([
-          { memcmp: { offset: 8, bytes: owner.toBase58() } },
-          { memcmp: { offset: 40, bytes: proposal.account.openbookFailMarket.toBase58() } },
-        ]);
-        setAllOrders({
-          [proposal.publicKey.toString()]: passOrders
```

Comment:
Similarly to market, orders are stored in a dictionary using the proposal public key as the index and should therefore be used at the top level in the AutocratContext in its current form.

However, I believe that open orders are more specific and it could make sense to keep them in a proposal context.

Reply:
removed dictionary